sqllogictest-engines/src/mysql.rs

```diff
@@ -0,0 +1,78 @@
+use std::process::Command;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use mysql_async::prelude::Queryable;
+use mysql_async::Value;
+use sqllogictest::{DBOutput, DefaultColumnType};
+
+type Result<T> = std::result::Result<T, mysql_async::Error>;
+
+/// Connection configuration. This is a re-export of [`mysql_async::Opts`].
+pub type MySqlConfig = mysql_async::Opts;
+
+/// MySQL engine based on the client from [`mysql_async`].
+pub struct MySql {
+    pool: mysql_async::Pool,
+}
+
+impl MySql {
+    /// Connects to the MySQL server with the given `config`.
+    pub async fn connect(config: MySqlConfig) -> Result<Self> {
+        let pool = mysql_async::Pool::new(config);
+        Ok(Self { pool })
+    }
+}
+
+#[async_trait]
+impl sqllogictest::AsyncDB for MySql {
+    type Error = mysql_async::Error;
+    type ColumnType = DefaultColumnType;
+
+    async fn run(&mut self, sql: &str) -> Result<DBOutput<Self::ColumnType>> {
+        let mut conn = self.pool.get_conn().await?;
+        let mut output = vec![];
+        let rows: Vec<mysql_async::Row> = conn.query(sql).await?;
+        for row in rows {
+            let mut row_vec = vec![];
+            for i in 0..row.len() {
+                // Since `query*` API in `mysql_async` is implemented using the MySQL text protocol,
+                // we can assume that the return value will be of type `Value::Bytes` or `Value::NULL`.
+                let value = row[i].clone();
+                let value_str = match value {
+                    Value::Bytes(bytes) => match String::from_utf8(bytes) {
+                        Ok(x) => x,
+                        Err(_) => "NULL".to_string(),
+                    },
+                    _ => "NULL".to_string(),
+                };
+                if value_str.is_empty() {
+                    row_vec.push("(empty)".to_string());
+                } else {
+                    row_vec.push(value_str);
+                }
+            }
+            output.push(row_vec);
+        }
+        if output.is_empty() {
+            Ok(DBOutput::StatementComplete(0))
```

Comment:
Is there anyway we can retrieve the affected row count?

Reply:
Yes, it can be obtained through `conn.affected_rows`, updated.